### PR TITLE
creating a new branch pulled from the margin rn material ui fork

### PR DESCRIPTION
**Pull request by rmitchellPitzer**
I think I finally was able to make a new pull request for this! I apologize to anyone who has a few notifications about reviewing the same pr and for making multiple pr's that were immediately deleted!
extractedText is now updated in the state. Before, clicking off the text input without pressing enter would have cleared the input as extractedText wasn't updated and didn't have any value. Now, extractedText is updated to contain the defaultValue, so it shouldn't clear when it shouldn't. 
The color is also updated as the text normally has a gray color that doesn't return, so that's been removed in favor of just normal text color.

### CODE REVIEW

File: src/components/field/index.js [hunk 1]
```diff
@@ -195,14 +201,20 @@ export default class TextField extends PureComponent {
   }
 
   componentDidMount() {
+    if(this.props.defaultValue){
+      this.setState({ text: this.props.defaultValue })
+    }
```
Comment (DT-1236) [inline src/components/field/index.js]: Did we actually need to keep this?

File: src/components/field/index.js [hunk 2]
```diff
@@ -498,9 +530,11 @@ export default class TextField extends PureComponent {
   }
 
   inputStyle() {
+
     let { fontSize, baseColor, textColor, disabled, multiline } = this.props
 
-    let color = disabled || this.isDefaultVisible() ? baseColor : textColor
+    let color = disabled || textColor
+    // let color = textColor
```
Comment (DT-1236) [inline src/components/field/index.js]: We can prolly get rid of this comment

File: src/components/field/index.js [hunk 3]
```diff
@@ -651,13 +692,14 @@ export default class TextField extends PureComponent {
         onContentSizeChange={this.onContentSizeChange}
         onFocus={this.onFocus}
         onBlur={this.onBlur}
-        value={value}
+        value={this.props.defaultValue}
```
Comment (DT-1236) [inline src/components/field/index.js]: This seems wrong. Doesn't this mean that the value of the `Input` won't ever update?

File: src/components/field/index.js [hunk 4]
```diff
@@ -324,74 +345,78 @@ export default class TextField extends PureComponent {
...
-    }
 
     this.focused = true
 
     this.startFocusAnimation()
     this.startLabelAnimation()
 
     if (!receivedFocus) {
-      this.setState({ receivedFocus: true, text: this.value() })
+      this.setState({ receivedFocus: true, text: this.value(), extractedText: this.value()})
```
Comment (DT-1236) [inline src/components/field/index.js]: I'm hoping to get some input from @amy-at-braid on this one. I feel like there should be a better way to get extracted text, but I'm also unwilling to expose it from the underlying mask input component. Any ideas?

**Review by DT-1236 — COMMENTED** [review]
Thanks for digging into this one. The main concern is breaking existing functionality/introducing other bugs. I got some questions that ought to be answered first. I'm not sure if they're actually breaking, so I can't make a hard decision on it just yet

File: src/components/field/index.js [hunk 5]
```diff
@@ -13,17 +13,20 @@ import styles from './styles'
 import TextInputMask from 'react-native-text-input-mask'
 
 function startAnimation(animation, options, callback) {
+
```
Comment (amy-at-braid) [inline src/components/field/index.js]: As a best practices thing, it's good to avoid extra whitespace changes since they make it harder to skim a code review. (Don't bother fixing it it here, just something for the future. I assume `prettier` just went nuts with this file.)

File: src/components/field/index.js [hunk 6]
```diff
@@ -324,74 +345,78 @@ export default class TextField extends PureComponent {
...
   onFocus(event) {
+
     let { onFocus, clearTextOnFocus } = this.props
     let { receivedFocus } = this.state
 
     if (typeof onFocus === 'function') {
       onFocus(event)
     }
 
-    if (clearTextOnFocus) {
```
Comment (amy-at-braid) [inline src/components/field/index.js]: Hm, is this change needed?

File: src/components/field/index.js (hunk 3 above)
Comment (amy-at-braid) [reply src/components/field/index.js]: Hm, yeah, I have the same question here.

File: src/components/field/index.js (hunk 4 above)
Comment (amy-at-braid) [reply src/components/field/index.js]: I'm a little confused here, shouldn't `extractedText` and `text` be different? I assume `text` is formatted whereas `extractedText` isn't, right? It's been a while since I've looked at this though.

**Comment by amy-at-braid** [comment]
Left a couple questions before I approve.

File: src/components/field/index.js (hunk 4 above)
Comment (rmitchellPitzer) [reply src/components/field/index.js]: text is formatted but extractedText isn't, so text would contain a "$" but extractedText would not

File: src/components/field/index.js (hunk 1 above)
Comment (rmitchellPitzer) [reply src/components/field/index.js]: Nope, that's been removed

File: src/components/field/index.js (hunk 3 above)
Comment (rmitchellPitzer) [reply src/components/field/index.js]: It seems to break when not using defaultValue, not sure what should be used here.

File: src/components/field/index.js (hunk 5 above)
Comment (rmitchellPitzer) [reply src/components/field/index.js]: Prettier I don't think is complaining about the file, will see if I can figure out enabling prettier for this file because I don't want white spaces everywhere in the file.

File: src/components/field/index.js (hunk 6 above)
Comment (rmitchellPitzer) [reply src/components/field/index.js]: No, added that back in.

File: src/components/field/index.js (hunk 4 above)
Comment (amy-at-braid) [reply src/components/field/index.js]: Hm, then why are they both equal to `this.value()` here? They should be different, no?

Comment (DT-1236) [reply src/components/field/index.js]: They should be, but I don't have any good ideas on how to apply the mask algorithm to the text and obtain extracted text properly. Do you have any ideas, @amy-at-braid

**Comment by rmitchellPitzer** [comment]
made some more changes, hopefully this is close to the end of this pr!

```
     if (key === 'defaultValue') {
         continue
       }
```

this block has been removed as it just seems to skip over the default value being passed in,

``` 
    let color = disabled || this.isDefaultVisible() ? baseColor : textColor
```

has been changed to 

```
    let color = disabled || textColor
```

as this just means a grey text color is seen only when first loading in and not again,
The call to setState in onFocus is now:

``` 
      this.setState({ receivedFocus: true, text: this.value(), extractedText: " " })
```
The reasoning behind this is that extractedText seemingly can be anything, as long as it is not null, and will work fine. I'm setting it to " " just to avoid any weird cases with text and extractedText having the same value.

**Review by DT-1236 — APPROVED** [review]
Changes seem very scoped and resolve the bug. Let's hope it doesn't introduce new bugs 🤞
